Aptoide/aptoide-client-v8
# Feature/asv 1092 donations implementation

**What does this PR do?**

This PR aims to add the ability to make donations to apps in aptoide. This can be done in the appview if the app is registred in the BDS. This meant addign 2 new elements to appview: a donations list and a call to action card after pressing install. These elements should only be shown if donations are available. To get the donations list a request is made to a v8 microservice which in turn meant a new user-agent interceptor had to be created. It can be reused for all calls to v8 web services. 

Once they press donate, users will see a donations dialog. This dialog is based on a fragment dialog and uses MVP. (NOTE: injection of members into this dialog are made from activity module due to the fact that our architecture takes care of the injection on a baseFragment class that cannot be extended by the dialog. It would be interesting to create a baseDialogFragment that takes care of this and would then be used as the base for all future dialogs). On clicking donate a request will be made to our web services to translate the package name into a wallet address, create a pending intent for the wallet with it and call it.

If the wallet isn't installed, clicking the appview donate button will show a dialog prompting you to install the wallet instead.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] DonateDialogFragment.java
- [ ] UserAgentInterceptorV8.java

**How should this be manually tested?**

Make sure you DO NOT have the wallet installed;
First, check editorial, highlighted, etc... (apps without bds billing) and make sure none show donation elements;
Afterwards go into the bds bundle/store/search and open an app. Confirm the donation elements are shown;
Press donate and verify the dialog for that wallet shows and navigates you to the wallet appview;
Install the wallet;
Go back to the appview and donate again;
Try to break the donations dialog;
Donate a value and see the flow resolve in the wallet;

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1092](<https://aptoide.atlassian.net/browse/ASV-1092>)

**Code Review Checklist**
